backend/src/test/java/de/sambam/gamenighttracker/service/UserServiceTest.java

```diff
@@ -31,17 +35,148 @@ public void listAllGamesTest() {
+        List<Game> list = userService.listAllGames("1");
 
         //THEN
-        assertTrue(listOptional.equals(Optional.of(List.of(
+        assertTrue(list.equals(List.of(
                 Game.builder()
                         .name("MauMau")
-                        .build()))));
+                        .build())));
+
+    }
+
+    @Test
+    @DisplayName("listAllSessions() should return all game sessions")
+    public void listAllSessionsTest() {
+        //GIVEN
+        when(userDb.findById("1")).thenReturn(Optional.of(User.builder()
+                .id("1")
+                .userName("admin")
+                .playedGames(List.of(
+                        Game.builder()
+                                .name("MauMau")
+                                .gameSessionList(List.of(
+                                        GameSession.builder()
+                                                .id("1")
+                                                .sessionState("DONE")
+                                                .winnerPlayerId("Sanne")
+                                                .build(),
+                                        GameSession.builder()
+                                                .id("2")
+                                                .sessionState("DONE")
+                                                .winnerPlayerId("Mario")
+                                                .build()
+                                ))
+                                .build(),
+                        Game.builder()
+                                .name("Monopoly")
+                                .gameSessionList(List.of(
+                                        GameSession.builder()
+                                                .id("3")
+                                                .sessionState("DONE")
+                                                .winnerPlayerId("Hanno")
+                                                .build()
+                                ))
+                                .build()))
+                .build()));
+
+        UserService userService = new UserService(userDb);
+
+        //WHEN
+        List<GameSession> gameSessionList = userService.listAllSessions("1");
+
+        //THEN
+        assertTrue(gameSessionList.equals(List.of(
+                GameSession.builder()
+                        .id("1")
+                        .sessionState("DONE")
+                        .winnerPlayerId("Sanne")
+                        .build(),
+                GameSession.builder()
+                        .id("2")
+                        .sessionState("DONE")
+                        .winnerPlayerId("Mario")
+                        .build(),
+                GameSession.builder()
+                        .id("3")
+                        .sessionState("DONE")
+                        .winnerPlayerId("Hanno")
+                        .build()
```

Comment:
you are using quite a lot of builder of your test class
you may want to consider extracting individual helper methods which return specific objects making your tests easier to read and reduce redundant code
(simple example)
```

private PlayerObject getPlayer() {return new PlayerObject;}

//later used in test method
public void testMethod() {
// Given
PlayerObject player = getPlayer();
}
```

Reply:
i'll try that, thanks! the tests have been exhausting, yeah ^^'